recipes/allegro5/all/conanfile.py

```diff
@@ -0,0 +1,313 @@
+            '''set(VORBIS_FOUND 1)
+               set(OGG_INCLUDE_DIR {})
+               set(VORBIS_INCLUDE_DIR {})
+               set(OGG_LIBRARIES {})
+               set(VORBIS_LIBRARIES {} {} {})
+               message("-- Using VORBIS from conan package")'''.format(
+                   ogg.package_folder.replace("\\","/") + "/include", vorbis.package_folder.replace("\\","/") + "/include", 
+                   ogg.package_folder.replace("\\","/") + "/lib/" + prefix + ogg.cpp_info.components["ogglib"].libs[0] + suffix,
+                   vorbis.package_folder.replace("\\","/") + "/lib/" + prefix + vorbis.cpp_info.components["vorbisfile"].libs[0] + suffix,
+                   vorbis.package_folder.replace("\\","/") + "/lib/" + prefix + vorbis.cpp_info.components["vorbismain"].libs[0] + suffix,
+                   ogg.package_folder.replace("\\","/") + "/lib/" + prefix + ogg.cpp_info.components["ogglib"].libs[0] + suffix))
+
+        if not self.settings.compiler == "Visual Studio":
+
+            # libalsa dependency
+            tools.replace_in_file(str(os.path.join(self.build_folder, "allegro5/addons/audio/CMakeLists.txt")), 
+                "pkg_check_modules(ALSA alsa)",
+                '''set(ALSA_FOUND 1)
+                   set(ALSA_INCLUDE_DIRS {})
+                   set(ALSA_LIBRARIES {})
+                   message("-- Using ALSA from conan package")'''.format(
+                       alsa.package_folder.replace("\\","/") + "/include", 
+                       alsa.package_folder.replace("\\","/") + "/lib/" + prefix + alsa.cpp_info.libs[0] + suffix))
+
+            # PulseAudio dependency
+            tools.replace_in_file(str(os.path.join(self.build_folder, "allegro5/addons/audio/CMakeLists.txt")), 
+                "pkg_check_modules(PULSEAUDIO libpulse-simple)",
+                '''set(PULSEAUDIO_FOUND 1)
+                   set(PULSEAUDIO_INCLUDE_DIRS {})
+                   set(PULSEAUDIO_LIBRARIES {})
+                   set(PULSEAUDIO_LIBRARY_DIRS {})
+                   message("-- Using PulseAudio from conan package")'''.format(
+                       pulseaudio.package_folder.replace("\\","/") + "/include", 
+                       pulseaudio.cpp_info.components["pulse"].libs[0],
+                       pulseaudio.package_folder.replace("\\","/") + "/lib/"))
+
+        # Disable specific compiler warnings
+        if self.settings.compiler == "Visual Studio":   
+            flags += " -DCMAKE_CXX_FLAGS=\"/wd4267\""   # possible loss of data warning
+        else:
+            flags += " -Wno-unused-variable -Wp,-w"
+
+        # Call cmake generate
+        path = Path(self.build_folder + "/allegro5/build")
+        path.mkdir(parents=True, exist_ok=True)
+        os.chdir(path)
+        self.run("cmake .. " + flags)
+
+    def build(self):
+        if self.settings.os == "Windows":
+            self.run("cd allegro5/build & cmake --build . --config RelWithDebInfo") # Build the project
```

Comment:
it doesn't seem right to hard-code build type here. better to use `self.settings.build_type`

Reply:
or just use CMake helper